### DISCUSSION TITLE
The audio stops when a looping video behind finishes. Media elements interlinked?

### DISCUSSION THREAD

**Discussion by h36ahmed** [katspaugh/wavesurfer.js#2981] [Q&A]
## Bug description
The audio continuously stops when the player is playing. I press Play button and then 3s later it stops and then I have to play again. I have tried multiple ways but I just can't find a solution. Here is the code snippet I have. I am using Next.js. I love this library - I would appreciate some help here. 

## Environment
 - Browser: Safari iOS
 - Software 16.1.1
 - Next.js
 - Wavesurfer 7 beta

## Minimal code snippet
```
import { useCallback, useEffect, useRef, useState } from 'react';
import type { WaveSurferOptions } from 'wavesurfer.js';
import WaveSurfer from 'wavesurfer.js';

import Button from '../Button/Button';

interface Props {
  url: string;
}

// WaveSurfer hook
const useWavesurfer = (containerRef: any, options: WaveSurferOptions) => {
  const [wavesurfer, setWavesurfer] = useState<WaveSurfer | null>(null);

  // Initialize wavesurfer when the container mounts
  // or any of the props change
  useEffect(() => {
    if (!containerRef.current) return;

    const ws = WaveSurfer.create({
      ...options,
      container: containerRef.current,
    });

    setWavesurfer(ws);

    // eslint-disable-next-line consistent-return
    return () => {
      ws.destroy();
    };
  }, [options, containerRef]);

  return wavesurfer;
};

// Create a React component that will render wavesurfer.
// Props are wavesurfer options.
const WaveSurferPlayer = (props: WaveSurferOptions) => {
  const containerRef = useRef<any>();
  const [isPlaying, setIsPlaying] = useState(false);
  const wavesurfer = useWavesurfer(containerRef, props);

  // On play button click
  const onPlayClick = useCallback(() => {
    if (wavesurfer?.isPlaying()) {
      wavesurfer?.pause();
    } else {
      wavesurfer?.play();
    }
  }, [wavesurfer]);

  // Initialize wavesurfer when the container mounts
  // or any of the props change
  useEffect(() => {
    if (!wavesurfer) return;

    setIsPlaying(false);

    const subscriptions = [
      wavesurfer.on('play', () => setIsPlaying(true)),
      wavesurfer.on('pause', () => setIsPlaying(false)),
    ];

    // eslint-disable-next-line consistent-return
    return () => {
      subscriptions.forEach((unsub) => unsub());
    };
  }, [wavesurfer]);

  return (
    <div className="flex w-1/2 flex-col items-center gap-4">
      <Button text={isPlaying ? 'PAUSE' : 'PLAY'} handleClick={onPlayClick} />
      <div
        id="waveform"
        ref={containerRef}
        className="mt-3 w-full"
        style={{ minHeight: '50px' }}
      />
    </div>
  );
};

const NextAudio = ({ url }: Props) => {
  const options = {
    barWidth: 3,
    cursorWidth: 1,
    container: '#waveform',
    height: 50,
    progressColor: '#ffffff',
    waveColor: '#707070',
    cursorColor: 'transparent',
    interact: false,
    url,
  };

  return <WaveSurferPlayer {...options} />;
};

export default NextAudio;

```

`
  "dependencies": {
    "framer-motion": "^10.12.16",
    "next": "^13.4.4",
    "next-seo": "^6.0.0",
    "next-sitemap": "^4.1.3",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-intersection-observer": "^9.5.1",
    "sharp": "^0.32.1",
    "swiper": "^9.4.1",
    "wavesurfer.js": "^7.0.0-beta.11"
  },
`

## Expected result
The audio plays without stopping 

## Obtained result
The audio file is a minute long and the player stops at least 5-8 times. Any chance you can take a look at this issue?

**Comment by h36ahmed** [comment]
Right, I investigated it further and now I have isolated the issue. The audio player works fine if it is the only media element on the page. I had a looping video playing behind that caused this issue. As soon as video finished, it would stop the audio in wavesurfer.js. Any reason why it would happen. Can those 2 items not affect each other? I am using Cloudinary Video Player..

**Comment by katspaugh** [comment]
Converting this to a discussion since there's no evidence it's a bug in wavesurfer.

I would start by debugging your React code. E.g. check when `NextAudio` is re-rendered. You're creating a new `options` object on each render which will destroy the wavesurfer instance every time.

**Reply by h36ahmed** [reply]
Ah possible - let me check

**Reply by katspaugh** [reply]
Actually I'm not sure it will cause a re-render of wavesurfer because the props are spread, it's not passing options as an object. Still worth checking.

**Reply by h36ahmed** [reply]
<img width="938" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/13081913/0e4a8612-36b7-45b9-b1ef-e765be88b232">
this what I see - rerendering is not an issue.

**Reply by h36ahmed** [reply]
Looks like there maybe an event fired from CloudinaryVideo Component that is captured by the wavesurfer.js and thus it pauses.

**Reply by katspaugh** [reply]
I don't know what CloudinaryVideo does behind the scenes but I strongly doubt it can affect wavesurfer. It has an internal media element that isn't even added to the DOM. It listens to events only from that internal media reference.

When it pauses, does the current time stay the same?

**Reply by h36ahmed** [reply]
yes, for wavesurfer, timer remains the same - the video keeps autoplaying

**Reply by h36ahmed** [reply]
<img width="1020" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/13081913/d4f26617-e9b3-4c49-881b-21906969fa98">
Got more into debugging and I see this that the player class is firing a pause event when the other video ends.

**Reply by katspaugh** [reply]
Like I said, it's impossible that the other video element affects the wavesurfer audio. It's completely isolated and not even added to the DOM. There's simply no way it can be accessed from outside other than by calling wavesurfer methods.
